### PR TITLE
Adding support for running the python version of ciceroscm

### PR DESCRIPTION
**Pull request by maritsandstad**
Adding support for running the python version of ciceroscm

- [ ] Tests added
- [ ] Documentation added
- [ ] Example added (in the documentation, to an existing notebook, or in a new notebook)
- [ ] Description in ``CHANGELOG.rst`` added (single line such as: ``(`#XX <https://github.com/openscm/openscm-runner/pull/XX>`_) Added feature which does something``)

### CODE REVIEW

**Comment by maritsandstad** [comment]
I'll try to clean up duplicate code and make a notebook example, so not done yet, but I think it's working in principle.

**Comment by maritsandstad** [comment]
I seem to be mainly failing from lack of magicc at the moment...

File: src/openscm_runner/adapters/ciceroscm_py_adapter/_compat.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+"""
+Handling of compatibility of fair imports with different states of installation
```
Comment (znicholls) [inline src/openscm_runner/adapters/ciceroscm_py_adapter/_compat.py:2]:
```suggestion
Handling of compatibility of CICEROSCM imports with different states of installation
```

File: src/openscm_runner/adapters/ciceroscm_py_adapter/ciceroscmpy.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        """
+        Get the CICEROSCM version being used by this adapter
+
+        Returns
+        -------
+        str
+            The CICEROSCM version id
+        """
+        # return ciceroscmpy.__version__
+        return "version-unknown"
```
Comment (znicholls) [inline src/openscm_runner/adapters/ciceroscm_py_adapter/ciceroscmpy.py:64]: Temporary or intended to merge as is?

File: src/openscm_runner/adapters/ciceroscm_py_adapter/cscmpy_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                (years, timeseries, unit,) = self.resultsreader.get_variable_timeseries(
+                    self.cscm.results, variable, self.sdatagetter
+                )
+                if isinstance(years, pd.DataFrame) and years.empty:  # pragma: no cover
+                    continue  # pragma: no cover
+                runs.append(
+                    ScmRun(
+                        pd.Series(timeseries, index=years),
+                        columns={
+                            "climate_model": "CICERO-SCM-PY",
```
Comment (znicholls) [inline src/openscm_runner/adapters/ciceroscm_py_adapter/cscmpy_wrapper.py:79]: Do you want it to have a different name or should it just be called CICERO-SCM (or CICEROSCM)?

File: tests/integration/test_cscmpy.py [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+            ),
+            out_config=None,
+        )
+        assert isinstance(res, ScmRun)
+        assert res["run_id"].min() == 1
+        assert res["run_id"].max() == 30040
+        assert (
+            res.get_unique_meta("climate_model", no_duplicates=True) == "CICERO-SCM-PY"
+        )
+        print(res.meta)
```
Comment (znicholls) [inline tests/integration/test_cscmpy.py:106]:
```suggestion
```

File: tests/integration/test_cscmpy.py [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+            scenarios=test_scenarios.filter(scenario="ssp126"),
+            output_variables=common_variables,
+        )
+
+        missing_vars = set(common_variables) - set(res["variable"])
+        if missing_vars:
+            raise AssertionError(missing_vars)
+
+    @pytest.mark.ciceroscm
+    def test_w_out_config(self, test_scenarios):
```
Comment (znicholls) [inline tests/integration/test_cscmpy.py:287]: So if you're missing any of the config, it doesn't run?

**Review by znicholls — APPROVED** [review]
lgtm, will take a look at CI now

**Comment by znicholls** [comment]
Alrighty @maritsandstad my attempts to give you access to secrets to make the CI work have failed so far. Can you make a branch in the openscm-runner repository and then create a new PR from that? That will give you access to the secrets while I try and work out a better solution (cc @lewisjared in case you have a better suggestion)

File: src/openscm_runner/adapters/ciceroscm_py_adapter/cscmpy_wrapper.py (hunk 3 above)
Comment (maritsandstad) [reply src/openscm_runner/adapters/ciceroscm_py_adapter/cscmpy_wrapper.py:79]: Asking the difficult questions...

File: setup.cfg [hunk 6]
```diff
@@ -59,9 +59,13 @@ fair =
 magicc =
     pymagicc >= 2.0.0, < 3
 
+ciceroscmpy =
+    ciceroscm @ git+https://github.com/ciceroOslo/ciceroscm.git
```
[the remark this answers is not in the document]
Comment (maritsandstad) [reply setup.cfg:63]: No, need to figure out how to put it there...

**Comment by maritsandstad** [comment]
I'll make the separate branch and fix other stuff.

File: src/openscm_runner/adapters/ciceroscm_py_adapter/ciceroscmpy.py (hunk 2 above)
Comment (maritsandstad) [reply src/openscm_runner/adapters/ciceroscm_py_adapter/ciceroscmpy.py:64]: Temporary

**Comment by maritsandstad** [comment]
Did it in here: #78 so this can likely be closed

**Comment by znicholls** [comment]
> Did it in here: #78 so this can likely be closed

Perfect thanks

File: setup.cfg (hunk 6 above)
[the remark this answers is not in the document]
Comment (znicholls) [reply setup.cfg:63]: If you make yourself a secret for pypi, you should be able to more or less copy the CI we use elsewhere (e.g. https://github.com/openscm/openscm-zenodo/blob/4448593ddd11086da6b371af14c90d0834314233/.github/workflows/ci-cd-workflow.yml#L119)

[the remark this answers is not in the document]
Comment (znicholls) [reply setup.cfg:63]: To make a secret, login to pypi, go to account settings then scroll down to 'API tokens' and hit 'Add API token'. Rest should be pretty clear I hope

[the remark this answers is not in the document]
Comment (maritsandstad) [reply setup.cfg:63]: Thanks, I'll see if I can't manage to do that... (Won't be the first time I'm shamelessly copying your code...)

File: tests/integration/test_cscmpy.py (hunk 5 above)
Comment (maritsandstad) [reply tests/integration/test_cscmpy.py:287]: Not sure I quite understand the question, this method is mainly checking what happens if you send the out_config parameter set to anything but None. If you do, it should throw an error. However, there are some required configs which I should maybe also be testing for...

Comment (znicholls) [reply tests/integration/test_cscmpy.py:287]: Ah ok got it

**Pull request closed without merging**